### ISSUE TITLE
str.join() reports a TypeError when called on symbolic input.

### ISSUE THREAD

**Issue by pschanely** [pschanely/CrossHair#8]
**Expected vs actual behavior**
The repro case below incorrectly fails with a TypeError saying "sequence item 0: expected str instance, SmtStr found".

str.join is actually quite problematic for CrossHair right now, because (unlike integers, list, etc) there is no good way to make a string-like object that doesn't inherit from string.

**To Reproduce**
```
def f(items: List[str]) -> str:
  ''' post: True '''
  return ', '.join(items)
```

**Comment by pschanely** [comment]
FWIW, I think a reasonable plan of attack here is to detect the problem when it happens and fall back to fuzzing with concrete strings.

**Comment by pschanely** [comment]
CrossHair now catches TypeErrors with strs that have "SmtStr found" and raises CrosshairUnsupported, which prevents the false alarm at least. It would still be better to be able to fuzz with concrete strings in these cases though.

**Comment by pschanely** [comment]
This was symbolically implemented in f1e8ce625e92e1e25e2540b1de2c94adc5ef50a8. Closing!
